src/utils/dimensionality_operations.jl

```diff
@@ -398,3 +431,60 @@ function tensor_cat(X::AbstractArray{T,N}, Y::AbstractArray{T,N}) where {T, N}
 end
 
 tensor_cat(X::Tuple{AbstractArray{T,N}, AbstractArray{T,N}}) where {T, N} = tensor_cat(X[1], X[2])
+
+"""
+    X_save, X = split_states(X, Z)
+
+ Split 1D vector in latent space back to states Zi. For non-conditional (single track) networks.
+
+ *Input*:
+
+ - `Y`, `Z`: ND input tensors, each of dimensions `nx` [x `ny` [x `nz`]] x `n_channel` x `batchsize`
+
+ *Output*:
+
+ - `X`: ND output tensor of dimensions `nx` [x `ny` [x `nz`]] x `n_channel*2` x `batchsize`
```

Comment:
I don't think that's the correct docstring. this is for tensor_split

Reply:
Now using your version of this function